Objects/unicodeobject.c

```diff
@@ -15708,10 +15708,8 @@ unicodeiter_next(unicodeiterobject *it)
     if (it->it_index < PyUnicode_GET_LENGTH(seq)) {
         int kind = PyUnicode_KIND(seq);
         const void *data = PyUnicode_DATA(seq);
-        Py_UCS4 chr = PyUnicode_READ(kind, data, it->it_index);
-        item = PyUnicode_FromOrdinal(chr);
-        if (item != NULL)
-            ++it->it_index;
+        Py_UCS4 chr = PyUnicode_READ(kind, data, it->it_index++);
```

Comment:
Note that the `PyUnicode_READ` macro reuse its arguments. I'm not sure if it is possible construct strings that misuse this misfeature, though.

Reply:
That makes the `it->it_index++` call a bit suspect. Maybe we should move it out of the macro.

Reply:
See the expanded macro, the index cannot be reused.

```c
((Py_UCS4) ((kind) == PyUnicode_1BYTE_KIND ? ((const Py_UCS1 *)(data))[(it->it_index++)] : ((kind) == PyUnicode_2BYTE_KIND ? ((const Py_UCS2 *)(data))[(it->it_index++)] : ((const Py_UCS4 *)(data))[(it->it_index++)] ) ))
```

Reply:
Even though the macro *happens* to work correctly, I would prefer not to have to rely on knowing that. I think we can trust the optimizer to do the right thing if we increment the index after the macro call.

Reply:
done